media/js/app/taxonomy/taxonomy.js

```diff
@@ -612,10 +546,48 @@
                             });
                         }
                     }
+                    //once we get to last iteration we save it all
+                     if (i == MAX - 1) {
```

Comment:
As a final save step, this section is probably better placed outside of the "onomy data" for loop.